src/Library/Plugin/Notice.php

```diff
@@ -0,0 +1,295 @@
+     * 
+     * @var bool
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $isUnread;
+
+    /**
+     * Constructor 
+     * 
+     * @param Plugin $plugin Plugin instance that this Notice belongs to
+     * @param array $notice An array of Notice data from plugin config
+     * @since SINCEVERSION
+     * 
+     */
+    public function __construct( Plugin $plugin, array $notice ) {
+        if ( false !== $this->alreadyExists( $notice['id'] ) ) {
+            $originalNotice = $this->getNoticeFromOptions( $notice['id'] );
+            $this->id       = $originalNotice->id;
+            $this->pageSlug = $originalNotice->pageSlug;
+            $this->plugin   = $originalNotice->plugin;
+            if ( version_compare( $originalNotice->version, $notice['version'], '!=' ) ) {
+                $this->version   = $notice['version'];
+                $this->isUnread = true;
+            }
+            else {
+                $this->version = $originalNotice->getVersion();
+                $this->isUnread = $originalNotice->isUnread;
+            }
+            $this->updateNoticeOption( $this );
+        } else {
+            $this->id        = $notice['id'];
+            $this->pageSlug  = $notice['page']; 
+            $this->version   = $notice['version'];
+            $this->isUnread  = true;
+            $this->plugin    = $plugin;
+        }
+    }
+
+	/**
+	 * Get notice ID
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getId() {
+        return $this->id;
+    }
+
+	/**
+	 * Set notice ID
+	 *
+	 * @param string $id Notice ID.
+     * @access private
+	 * @since SINCEVERSION
+	 *
+	 */
+	private function setId( $id ) {
+		$this->id = $id;
+	}
+
+	/**
+	 * Get slug of the page this feature is on.
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getPageSlug() {
+        return $this->pageSlug;
+    }
+
+	/**
+	 * Set pageSlug
+	 *
+	 * @param string $pageSlug Notice Page.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setPageSlug( string $pageSlug ) {
+		$this->pageSlug = $pageSlug;
+	}
+
+	/**
+	 * Get version of plugin this notice was added on
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getVersion() {
+        return $this->version;
+    }
+
+	/**
+	 * Set version of plugin this notice was added on
+	 *
+	 * @param string $version Version of plugin this notice was added on.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setVersion( string $version ) {
+		$this->version = $version;
+	}
+
+    /**
+     * Maybe Show
+     * 
+     * Returns true if this plugin's version number is
+     * greater than the first installed version, and feature's version number is 
+     * greater than or equal to current version
+     * @return bool
+     * @since SINCEVERSION
+     */
+    
+    public function maybeShow() {
+        $pluginVersion = $this->plugin->getPluginData()['Version'];
```

Comment:
Generally we want to align the equals signs if the lines are touching.

```
        $pluginVersion     = $this->plugin->getPluginData()['Version'];
        $versionIsNotFirst = $this->plugin->firstVersionCompare( $pluginVersion, '<' );
        $featureIsNewer    = version_compare( $this->version, $pluginVersion , '>=' );
```

If the lines are not touching, you don't need to do this.
```
        $pluginVersion = $this->plugin->getPluginData()['Version'];

        $versionIsNotFirst = $this->plugin->firstVersionCompare( $pluginVersion, '<' );

        $featureIsNewer = version_compare( $this->version, $pluginVersion , '>=' );
```